From faae9556f78e5bde5f5b9c46382c137a4901cf00 Mon Sep 17 00:00:00 2001
From: Ofer Heifetz <oferh@marvell.com>
Date: Mon, 15 May 2017 11:05:38 +0300
Subject: [PATCH 1003/1345] fix: crypto: inside-secure: fix more EIP197 driver
 DMA handling

commit  e74f212ebd09b867d874bb2e159a7b3b9a3aa6dd from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

According to a review received for EIP197 driver:
Drivers which use DMA API calls must use only DMA addresses
and not physical addresses as used now in EIP197 driver
(when using physical addresses there's no guarantee that
the physical addresses will give the expected result).

Change-Id: I0d270ff42118e88aab0b14a91f88cd6d431d6183
Signed-off-by: Ofer Heifetz <oferh@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/39478
Reviewed-by: Igal Liberman <igall@marvell.com>
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/39528
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/crypto/inside-secure/hash.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/crypto/inside-secure/hash.c b/drivers/crypto/inside-secure/hash.c
index add6e2f..c77b88d 100644
--- a/drivers/crypto/inside-secure/hash.c
+++ b/drivers/crypto/inside-secure/hash.c
@@ -267,7 +267,7 @@ static int safexcel_ahash_send(struct crypto_async_request *async, int ring,
 
 	/* Add a result descriptor */
 	rdesc = safexcel_add_rdesc(priv, ring, 1, 1,
-				   dma_to_phys(priv->dev, ctx->base.result_dma),
+				   ctx->base.result_dma,
 				   req->state_sz);
 	if (IS_ERR(rdesc)) {
 		ret = PTR_ERR(rdesc);
-- 
1.7.9.5

